### PR TITLE
XCAROUSEL-2: the carousel image don't follows the carousel dimensions…

### PR DESCRIPTION
**Pull request by KebirYacine**
… & image has low quality

### CODE REVIEW

File: src/main/resources/XCarouselCode/XCarouselSheet.xml [hunk 1]
```diff
@@ -163,7 +163,7 @@
               #set ($itemClass = "item slide_${object.number}")
             #end
             &lt;div class="${itemClass}"&gt;
-              &lt;img src="$!{imageUrl}?height=$!{imageHeight}" alt=''&gt;
+              &lt;img src="$!{imageUrl}" alt="$!{object.getProperty('title').value}"&gt;
```
Comment (Enygma2002) [inline src/main/resources/XCarouselCode/XCarouselSheet.xml]: Always use `$escapetool.xml` when printing user-entered content into html. In this case, the `alt` attribute value needs escaping.

**Comment by KebirYacine** [comment]
$escapetool.xml added thank you Edy for your comment

**Comment by IdentityOverflow** [comment]
I believe that this changes would indeed make the image fit the carousel, but they would also distort the image if only the height would be set for example. So I would suggest centering the image instead, wdyt?

**Comment by KebirYacine** [comment]
Yes a agree with you for this

File: src/main/resources/XCarouselCode/XCarouselSheet.xml [hunk 2]
```diff
@@ -890,15 +894,17 @@ colorPikerOn();</code>
...
   #if("$!{carouselObj.getProperty('carouselWidth').value}" == '')
     #set($carouselImageWidth = '100%')
   #else 
     #set($carouselImageWidth = ${carouselWidth})
   #end
-  $carouselID .carousel-inner>.item>img, $carouselID .carousel-inner>.item>a>img {
-    width: ${carouselImageWidth};
-    height: ${carouselHeight};
+  $carouselID .item-inner img {
+    max-width: ${carouselImageWidth};
```
Comment (IdentityOverflow) [inline src/main/resources/XCarouselCode/XCarouselSheet.xml:904]: Firefox ignores max-width in certain situations, and in this case, if the image has a bigger width than the screen, it causes the image to not be fully displayed if the max-width is 100%.

File: src/main/resources/XCarouselCode/XCarouselSheet.xml [hunk 3]
```diff
@@ -919,6 +925,19 @@ colorPikerOn();</code>
       color: ${textColor};
       font-size: $!{textSize};
     }
+    $carouselID .item.slide_${slideObj.number} {
+      height: ${carouselHeight};
+    }
+    $carouselID .slide_${slideObj.number} .item-inner {
+      display: table;
```
Comment (IdentityOverflow) [inline src/main/resources/XCarouselCode/XCarouselSheet.xml:932]: A "table-layout: fixed;" here should fix the problem above.

**Pull request merged**